servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -200,16 +239,15 @@ public void abortBlock(long userId, long blockId) throws AlreadyExistsException,
   public void requestSpace(long userId, long blockId, long additionalBytes)
       throws NotFoundException, OutOfSpaceException, IOException, AlreadyExistsException,
       InvalidStateException {
-    // TODO: Change the lock to read lock and only upgrade to write lock if necessary
-    mEvictionLock.writeLock().lock();
-    try {
-      TempBlockMeta tempBlockMeta = mMetaManager.getTempBlockMeta(blockId);
-      freeSpaceInternal(userId, additionalBytes, tempBlockMeta.getBlockLocation());
-      // Increase the size of this temp block
-      mMetaManager.resizeTempBlockMeta(tempBlockMeta, tempBlockMeta.getBlockSize()
-          + additionalBytes);
-    } finally {
-      mEvictionLock.writeLock().unlock();
+    for (int i = 0; i < MAX_RETRIES; i ++) {
+      Pair<Boolean, BlockStoreLocation> requestResult =
+          requestSpaceInternal(blockId, additionalBytes);
+      if (requestResult.getFirst()) {
+        return;
+      }
+      if (i < MAX_RETRIES - 1) {
+        freeSpaceInternal(userId, additionalBytes, requestResult.getSecond());
```

Comment:
Do we want to retry if `requestSpaceInternal` or `freeSpaceInternal` throws an exception? If so,   we need catch the exceptions except for the last try.